baseobject/testing/test_all_objects.py

```diff
@@ -0,0 +1,789 @@
+
+        generator_dict = {}
+        for var, gen in zip(fixts, gens):
+            generator_dict[var] = getattr(self, gen)
+
+        return generator_dict
+
+    def is_excluded(self, test_name, est):
+        """Shorthand to check whether test test_name is excluded for object est."""
+        if self.excluded_tests is None:
+            return []
+        else:
+            return test_name in self.excluded_tests.get(est.__name__, [])
+
+    # the following functions define fixture generation logic for pytest_generate_tests
+    # each function is of signature (test_name:str, **kwargs) -> List of fixtures
+    # function with name _generate_[fixture_var] returns list of values for fixture_var
+    #   where fixture_var is a fixture variable used in tests
+    # the list is conditional on values of other fixtures which can be passed in kwargs
+
+    def _generate_object_class(self, test_name, **kwargs):
+        """Return object class fixtures.
+
+        Fixtures parametrized
+        ---------------------
+        object_class: object inheriting from BaseObject
+            ranges over all object classes not excluded by self.excluded_tests
+        """
+        object_classes_to_test = [
+            est for est in self._all_objects() if not self.is_excluded(test_name, est)
+        ]
+        object_names = [est.__name__ for est in object_classes_to_test]
+
+        return object_classes_to_test, object_names
+
+    def _generate_object_instance(self, test_name, **kwargs):
+        """Return object instance fixtures.
+
+        Fixtures parametrized
+        ---------------------
+        object_instance: instance of object inheriting from BaseObject
+            ranges over all object classes not excluded by self.excluded_tests
+            instances are generated by create_test_instance class method
+        """
+        # call _generate_object_class to get all the classes
+        object_classes_to_test, _ = self._generate_object_class(test_name=test_name)
+
+        # create instances from the classes
+        object_instances_to_test = []
+        object_instance_names = []
+        # retrieve all object parameters if multiple, construct instances
+        for est in object_classes_to_test:
+            all_instances_of_est, instance_names = est.create_test_instances_and_names()
+            object_instances_to_test += all_instances_of_est
+            object_instance_names += instance_names
+
+        return object_instances_to_test, object_instance_names
+
+    # this is executed before each test instance call
+    #   if this were not executed, object_instance would keep state changes
+    #   within executions of the same test with different parameters
+    @pytest.fixture(scope="function")
+    def object_instance(self, request):
+        """object_instance fixture definition for indirect use."""
+        # esetimator_instance is cloned at the start of every test
+        return request.param.clone()
+
+
+class QuickTester:
```

Comment:
Are you flexible on the name? Maybe TesterMixin? I'm going to approve this, but maybe we include a quick PR that updates the name before V0.1.0 release?

Reply:
Mainly I'm thinking that `QuickTester` implies that the primary benefit is speed or that this was done as a quick/intermediate output (but goal of package is well thought out testing interface, so we know that is not the case)